### PR TITLE
Add vsphere-problem-detector repo

### PR DESCRIPTION
**Pull request by jsafrane**
Add new repo https://github.com/openshift/vsphere-problem-detector + its image and unit test. In the future, the image has potential to block / break OCP on vSphere, so e2e-test that.

### CODE REVIEW

**Comment by jsafrane** [comment]
vsphere errors are not related to the new repo - the new image is not installed (yet).

File: core-services/template-deprecation/_allowlist.yaml [hunk 1]
```diff
@@ -1789,6 +1789,9 @@ templates:
         pull-ci-openshift-sdn-release-4.8-e2e-aws-multitenant:
           generated: false
           kind: presubmit
+        pull-ci-openshift-vsphere-problem-detector-master-e2e-aws:
+          generated: true
+          kind: presubmit
```
Comment (bbguimaraes) [inline core-services/template-deprecation/_allowlist.yaml:1794]: It doesn't look like this is necessary.

Comment (jsafrane) [reply core-services/template-deprecation/_allowlist.yaml:1794]: It was generated by make new-repo. Shall I remove it?

Comment (jsafrane) [reply core-services/template-deprecation/_allowlist.yaml]: I read README and it indeed seems to be unnecessary. Removed.

Comment (jsafrane) [reply core-services/template-deprecation/_allowlist.yaml]: I read README and it indeed seems to be unnecessary. Removed.

Comment (bbguimaraes) [reply core-services/template-deprecation/_allowlist.yaml]: Hmm, I'll check that.

**Pull request merged**